swim/join_sender.go

```diff
@@ -162,6 +203,15 @@ func (j *joinSender) parseHosts(hostports []string) {
 	}
 }
 
+func newDelayOpts() *delayOpts {
+	return &delayOpts{
```

Comment:
Would it make sense to not change the delayOpts after they are passed into the joiner and reuse the defaults object? Currently I only see changes made to `joinsender.delayOpts` during the tests which could also be passed in via the `newJoinSender`.